src/d3/hapticplot.d3.ts

```diff
@@ -108,12 +105,54 @@ export class Hapticplot{
    * @param hoverColor The color the point takes on while hovered
    * @param size The radius of the point being hovered
    */
-  private onHoverStart(point, hapticIntensity, hoverColor, size){
+  private onHoverStart(datum, point, hoverColor){
     d3.select(point)
       .attr('color', hoverColor);
     if (point.components?.sound?.isPlaying){
       point.components.sound.stopSound();
     }
+    d3.select(d3.event.detail?.hand)
+    .on('abuttonup',  (d, i, g) => this.speakPosition(datum, point))
+    .on('xbuttonup',  (d, i, g) => this.speakPosition(datum, point));
+  }
+
+  /**
+   * Reads out the position of the given point in spoken audio, and resets its sound component to
+   * the appropriate marimba note audio upon completion
+   * @param datum The datum from which the given points position and audio is generated
+   * @param point A data point entity
+   */
+  private speakPosition(datum, point){
```

Comment:
I've just noticed that a bunch of private methods in this file don't have any types in the params.  I actually thought TS required that, so I wasn't checking, but they should all have types.

Reply:
I added them! Doing so revealed a secret any cast when I was accessing the sound component, so I added an interface for the Sound type extending the generic Component type